ghga_service_chassis_lib/object_storage_dao_testing.py

```diff
@@ -70,6 +72,58 @@ def upload_file(presigned_url: PresignedPostURL, file_path: Path, file_md5: str)
         response.raise_for_status()
 
 
+def calc_part_size(file_path: Path, n_parts: int) -> int:
+    """Calculate the part size in bytes for the given file when desiring the specified number of
+    parts.
+    Please note, the part size must be at least 25 min"""
+    file_size = os.path.getsize(file_path)
+    return max(math.ceil(file_size / n_parts), 50 * pow(10, 6))
+
+
+def multipart_upload_file(
+    storage_dao: ObjectStorageDao,
+    bucket_id: str,
+    object_id: str,
+    file_path: Path,
+    n_parts: int,
+) -> None:
+    """Uploads the test file to the specified URL"""
+    print(f" - initiate multipart upload for test object {object_id}")
+    upload_id = storage_dao.init_mulitpart_upload(
+        bucket_id=bucket_id, object_id=object_id
+    )
+
+    part_size = calc_part_size(file_path=file_path, n_parts=n_parts)
```

Comment:
Didn't we agree on a fixed part size of 16 MiB? So, why are we calculating part sizes here?

Reply:
This code is only for testing, so is independent of our convention. However, I see your point, maybe instead of specifying the number of parts, we always specify the anticipated part size. Will change the interface accordingly.

Reply:
Done